File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/CheckTender.cs
#region "Imports"

using PayPal.Payments.Common.Utility;
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for Check tender related information.
	/// </summary>
	/// <remarks>CheckPayment is the Payment device associated with
	/// this tender type.
	/// <seealso cref="CheckPayment"/>
	/// </remarks>
	public sealed class CheckTender : BaseTender
	{

		#region "Constructors"

		/// <summary>
		/// Constructor for CheckTender
		/// </summary>
		/// <param name="Check">Check Payment object.</param>
		/// <remarks>This constructor is used to create a CheckTender
		///  with CheckPayment as the payment device</remarks>
		///  <example>
		///  <code lang="C#" escaped="false">
		///		.............
		///		//Check is the CheckPayment object.
		///		.............
		///		
		///		CheckTender Tender = new CheckTender(Check);
		///		
		///		..............
		///  </code>
		///  <code lang="C#" escaped="false">
		///		.............
		///		'Check is the CheckPayment object.
		///		.............
		///		
		///		Dim Tender As CheckTender = new CheckTender(Check)
		///		
		///		..............
		///  </code>
		///  </example>
		/// <seealso cref="CheckPayment"/>
		public CheckTender(CheckPayment Check) : base(PayflowConstants.TENDERTYPE_TELECHECK, Check)
		{
		}

		#endregion
	}
}

File: java/src/sdk/dataobjects/paypal/payflow/CheckTender.java
/*
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package paypal.payflow;

/**
 * Used for Check tender related information.
 * CheckPayment the Payment devices associated with this tender type.
 * {@link CheckPayment}
 */
public final class CheckTender extends BaseTender {
    /**
     * This constructor is used to create a CheckTender
     * with CheckPayment as the payment device
     *
     * @param check CheckTender
     * @paypal.sample Maps to Payflow Parameter: TENDER
     * <p/>
     * .............
     * //check is the populated CheckPaymentobject.
     * .............
     * <p/>
     * //Create the Tender object
     * CheckTender tender = new CardTender(check);
     */
    public CheckTender(CheckPayment check) {
        super(PayflowConstants.TENDERTYPE_TELECHECK, check);
    }

}